solutions/standard/main.tf

```diff
@@ -56,66 +55,78 @@ module "private_secret_engine" {
   source                 = "terraform-ibm-modules/secrets-manager-private-cert/ibm"
   version                = "1.3.1"
-  cert_name              = "${var.prefix}-cts-vpn-private-cert"
+  cert_name              = var.prefix != null ? "${var.prefix}-cts-vpn-private-cert" : "cts-vpn-private-cert"
   cert_description       = "an example private cert"
   cert_template          = var.certificate_template_name
-  cert_secrets_group_id  = module.secrets_manager_group.secret_group_id
+  cert_secrets_group_id  = module.secrets_manager_group[0].secret_group_id
   cert_common_name       = var.cert_common_name
-  secrets_manager_guid   = local.sm_guid
-  secrets_manager_region = local.sm_region
+  secrets_manager_guid   = module.existing_sm_crn_parser.service_instance
+  secrets_manager_region = module.existing_sm_crn_parser.region
   providers = {
     ibm = ibm.ibm-sm
   }
 }
 
+module "existing_secrets_manager_cert_crn_parser" {
+  count   = var.existing_secrets_manager_cert_crn != null ? 1 : 0
+  source  = "terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"
+  version = "1.0.0"
+  crn     = var.existing_secrets_manager_cert_crn
+}
+
 ##############################################################################
-# Deploy client-to-site in a dedicated subnet in the management Landing Zone VPC
+# Deploy client-to-site in a dedicated subnet in the VPC
 ##############################################################################
-
 locals {
-  zone_1 = var.vpn_zone_1 != null ? var.vpn_zone_1 : "${var.region}-1" # hardcode to first zone in region
-  zone_2 = var.vpn_zone_2 != null ? var.vpn_zone_2 : "${var.region}-2" # hardcode to second zone in region
+  existing_vpc_id = module.existing_vpc_crn_parser.resource
+  subnets = [
+    for subnet in data.ibm_is_vpc.existing_vpc_data[0].subnets :
+    subnet.id if can(regex(join("|", var.existing_subnet_names), subnet.name))
+  ]
+
+  subnet_ids = length(var.existing_subnet_names) > 0 ? local.subnets : var.vpn_subnet_cidr_zone_2 != null ? [ibm_is_subnet.client_to_site_subnet_zone_1[0].id, ibm_is_subnet.client_to_site_subnet_zone_2[0].id] : [ibm_is_subnet.client_to_site_subnet_zone_1[0].id]
+  zone_1     = var.vpn_zone_1 != null ? var.vpn_zone_1 : "${var.region}-1" # hardcode to first zone in region
+  zone_2     = var.vpn_zone_2 != null ? var.vpn_zone_2 : "${var.region}-2" # hardcode to secod zone in region
+  target_ids = [module.vpn.vpn_server_id]
 }
 
+module "existing_vpc_crn_parser" {
+  source  = "terraform-ibm-modules/common-utilities/ibm//modules/crn-parser"
+  version = "1.0.0"
+  crn     = var.existing_vpc_crn
+}
+
+data "ibm_is_vpc" "existing_vpc_data" {
+  count      = var.adjust_existing_vpc_acls && local.existing_vpc_id != null ? 1 : 0
+  identifier = local.existing_vpc_id
+}
 
 resource "ibm_is_vpc_address_prefix" "client_to_site_address_prefixes_zone_1" {
   count = length(var.existing_subnet_names) > 0 ? 0 : 1
   name  = "${var.prefix}-client-to-site-address-prefixes-1"
```

Comment:
prefix is optional, so this would break if no prefix was passed (there might be other places in code that also has the issue - please check)